src/XRP/issued-currency-client.ts

```diff
@@ -878,6 +882,174 @@ export default class IssuedCurrencyClient {
     )
   }
 
+  /**
+   * Send a cross-currency payment. Note that the source and destination currencies cannot both be XRP.
+   * The XRPL is queried for viable paths as part of transaction construction.  If no paths are found,
+   * an error is thrown and no transaction is submitted.
+   *
+   * @see https://xrpl.org/cross-currency-payments.html
+   *
+   * @param sender The wallet from which currency will be sent, and that will sign the transaction.
+   * @param destination The address of the payment's recipient, encoded as an X-Address.
+   * @param maxSourceAmount Highest amount of source currency this transaction is allowed to cost, including transfer fees,
+   *               exchange rates, and slippage. Does not include the XRP destroyed as a cost for submitting the transaction.
+   *               For XRP, specify amount as a string of drops.
+   *               For issued currencies, specify an Issued Currency object with currency, issuer, and value fields.
+   * @param deliverAmount The amount of currency to deliver. For XRP, specify amount as a string of drops.  For issued currencies,
+   *               specify an Issued Currency object with currency, issuer, and value fields.
+   */
+  public async sendCrossCurrencyPayment(
+    sender: Wallet,
+    destination: string,
+    maxSourceAmount: string | IssuedCurrency,
+    deliverAmount: string | IssuedCurrency,
+  ): Promise<TransactionResult> {
+    if (!XrpUtils.isValidXAddress(destination)) {
+      throw new XrpError(
+        XrpErrorType.XAddressRequired,
+        'Destination address must be in X-address format.  See https://xrpaddress.info/.',
+      )
+    }
+
+    // TODO: verify address formats for issuers in IssuedCurrency specifications.  They need to be classic addresses.
+    //        Do we want to take them in as classic addresses?? Or force them to be X-addresses like elsewhere?
+    // TODO: (acorso) we don't need to convert back to a classic address once the ripple-binary-codec supports X-addresses for issued currencies.
+
+    // Both source amount and deliver amount can't both be XRP:
+    if (
+      typeof deliverAmount == 'string' &&
+      typeof maxSourceAmount == 'string'
+    ) {
+      throw new XrpError(
+        XrpErrorType.InvalidInput,
+        'A Cross Currency payment should involve at least one issued currency.',
+      )
+    }
```

Comment:
Perhaps add another check on whether the deliver and maxSource currencies are the same or not